index.src.html

```diff
@@ -564,15 +606,47 @@ <h4 id="algorithm-store" algorithm>Store a `Credential`</h4>
 
     4.  Return a {{Promise}} which resolves with the result of executing |credential|'s
         [=interface object=]'s {{Credential/[[Store]](credential)}} internal method on |credential|.
-  </ul>
+  </ol>
+
+  <h4 id="algorithm-create" algorithm>Create a `Credential`</h4>
+
+  The <dfn abstract-op>Create a `Credential`</dfn> algorithm accepts a {{CredentialCreationData}}
+  (|data|), and returns a {{Promise}} which resolves with a {{Credential}} if one can be created
+  using the options provided, or with `null` if not.
+
+  <ol class="algorithm">
+    1.  Let |settings| be the <a>current settings object</a>
+
+    2.  Assert: |settings| is a [=secure context=].
+
+    3.  Let |interfaces| be the set of |data|' <a>relevant credential interface objects</a>.
+
+    4.  Return a {{Promise}} rejected with `NotSupportedError` if any of the following statements
+        are true:
+
+        1.  |settings| does not have a [=environment settings object/responsible document=].
+
+        2.  |settings|' [=environment settings object/responsible document=] is not the
+            [=active document=] in a [=top-level browsing context=].
+
+        3.  |interfaces|' [=list/size=] is greater than 1.
+
+            Note: It may be reasonable at some point in the future to loosen this restriction, and
+            allow the user agent to help the user choose among one of many potential credential
+            types in order to support a "sign-up" use case. For the moment, though, we're punting
+            on that by restricting the dictionary to a single entry.
+
+    5.  Return a {{Promise}} which resolves with the result of executing |interfaces|[0]'s
```

Comment:
Since [[Create]] can return `null`, `create()` currently returns a `Promise<Credential?>` instead of the non-null form that's currently written.